portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,31 +22,72 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.FetchOptions;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   
-  private ArrayList<String> quotes;
+  private List<String> comments = new ArrayList<>(); 
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    quotes = new ArrayList<>();
-    quotes.add("beep boop - ben lin");
-    quotes.add("They told me computers could only do arithmetic -Grace Hopper. I told them I cannot do arithmetic - Ben Lin");
-    quotes.add("hydrate or diedrate - Ben Lin");
-
-    // Convert the quotes to JSON using Gson library
-    String json = new Gson().toJson(quotes);
-
+    
     // Send the JSON as the response
     response.setContentType("application/json;");
-    response.getWriter().println(json);
     
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Message").addSort("timestamp", SortDirection.DESCENDING);
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(10));
+    // add a limit
+
+    List<String> commentsPopulate = new ArrayList<>();
+
+    for (Entity entity : results) {
+      String text = (String) entity.getProperty("text");
+      String email = (String) entity.getProperty("email");
+      commentsPopulate.add(email + ": " + text);
+    }
+
+    String json = new Gson().toJson(commentsPopulate);
+    response.getWriter().println(json);
+
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  
+    // Get the input from the form and add it to json on /data
```

Comment:
Continuing on comments: here isn't the best place to say 'add it to json', since it's not being converted to JSON. Arguably the other 2 parts aren't needed either since 'request.getParameter' should be clear, and this class is a handler for '/data'. It's up to you, though, feel free to keep those if you think it helps remind readers

Reply:
Thanks